solutions/src/main/java/ru/sberbank/school/task11/Streams.java

```diff
@@ -0,0 +1,179 @@
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+/**
+ * Облегченная версия класса {@link Stream}
+ */
+public class Streams<T> {
+    private final List<T> list = new ArrayList<>();
+
+    private final List<Function <List<T>, List<?>>> functions = new ArrayList<>();
+
+    private Streams(@NonNull T... elements) {
+        list.addAll(Arrays.asList(elements));
+    }
+
+    private Streams(@NonNull Collection<T> elements) {
+        list.addAll(elements);
+    }
+
+    /**
+     * Принимает на вход массив элементов и возвращает стрим построенный на основе этих элементов.
+     *
+     * @param elements входные элементы
+     * @return стрим элементов из elements
+     */
+    public static <T> Streams<T> of(@NonNull T... elements) {
+        return new Streams<>(elements);
+    }
+
+    /**
+     * Принимает на вход коллекцию элементов и возвращает стрим построенный на основе этих элементов.
+     *
+     * @param elements входные элементы
+     * @return стрим элементов из elements
+     */
+    public static <T> Streams<T> of(@NonNull Collection<T> elements) {
+        return new Streams<>(elements);
+    }
+
+    /**
+     * Фильтрация элементов по заданному правилу. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Иными словами, этот метод оставляет в коллекции только те элементы, которые удовлетворяют условию в лямбде.
+     * <p>
+     * Аналог Stream#map(Function).
+     * <p>
+     * Intermediate операция.
+     *
+     * @param object - правило фильтрации элементов.
+     * @return стрим
+     */
+    public Streams<T> filter(@NonNull Predicate<T> object) {
+        functions.add((toFilter) -> {
+            List<T> result = new ArrayList<>();
+            for (T it : toFilter) {
+                if (object.test(it)) {
+                    result.add(it);
+                }
+            }
+            return result;
+        });
+        return this;
+    }
+
+    /**
+     * Преобразование элементов в какие-то другие элементы. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Например, данный метод должен уметь извлекать какие-то поля из объектов исходного типа (если это возможно).
+     * <p>
+     * Аналог Stream#map.
+     * <p>
+     * Intermediate операция.
+     *
+     * @param function - правило траснформации элементов.
+     * @return стрим
+     */
+    @SuppressWarnings("unchecked")
+    public <R> Streams<R> transform(@NonNull Function<T, R> function) {
+        functions.add((toTransform) -> {
+            List<R> result = new ArrayList<>();
+            for (T it : toTransform) {
+                result.add(function.apply(it));
+            }
+            return result;
+        });
+        return (Streams<R>) this;
+    }
+
+    /**
+     * Сортировка элементов. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Данный метод должен уметь сортировать элементы по заданному правилу.
+     * <p>
```

Comment:
&lt;p&gt; tag should be preceded with an empty line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1556847699>*